Asterix/optics/propagation_functions.py

```diff
@@ -165,17 +225,50 @@ def mft(image,
             norm0 = np.sqrt(nbresx * nbresy / dim_input_x / dim_input_y / dim_output_x / dim_output_y)
         sign_exponential = 1
 
-    AA = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(uu0, xx0))
-    BB = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(xx1, uu1))
-    result = norm0 * np.matmul(np.matmul(AA, image), BB)
+    AA = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(uu0, xx0)).astype('complex64')
+    BB = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(xx1, uu1)).astype('complex64')
 
-    return result
+    if returnAABB:
+        return AA, BB, norm0
 
+    return mat_mult_mft(image.astype('complex64'), AA, BB, norm0)
 
-def prop_fresnel(pup, lam, z, rad, prad, retscale=0):
+
+def mat_mult_mft(image, AA, BB, norm0):
+    """Perform Matrix multiplication for MFT. It is be done separatly in to
+    allowed to be sped up by GPU (maybe). I tried using numba for this function
+    to save time but no improvements.
```

Comment:
I don't quite understand the second sentence in this docstring. Would you mind clarifying?